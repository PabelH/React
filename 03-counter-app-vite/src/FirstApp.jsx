// function App() {
//     return <h1>Hello Pabello</h1>
// }

//using rafc
// import React from 'react'

const FirstApp = () => {
  return (
    <div>Hello Pabelon </div>
  )
}

export default FirstApp



// export default App;
