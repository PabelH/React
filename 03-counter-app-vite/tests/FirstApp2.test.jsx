import { render, screen } from "@testing-library/react";
import FirstApp from "../src/FirstApp";


describe('Test in <FirstApp />', () => {

    const title = 'Hi MF'
    const subTitle = 'No SubtitleF'

    test('this test must match the snapshot"',()=>{

        const {container} = render(<FirstApp title={ title }/>);
        expect( container ).toMatchSnapshot();

    });

    test('this test should return "Hi MF"',()=>{

        render(<FirstApp title={ title }/>);
        expect( screen.getByText(title) ).toBeTruthy();
        //If you want it not to exist
        //expect( screen.getByText(title) ).not.toBeTruthy();
        //screen.debug();
        
    });

    test('this test should return the title in a h1 tag',()=>{

        render(<FirstApp title={ title }/>);
        expect( screen.getByRole('heading',{ level: 1 }).innerHTML ).toContain( title );
        
        
    });

    test('this test should show the subtitle sent by props',()=>{

        render(<FirstApp 
            title={ title }
            subTitle={ subTitle }
            />);
            expect( screen.getAllByText(subTitle).length ).toBe(2);

    
    
    
    });
    
  
});