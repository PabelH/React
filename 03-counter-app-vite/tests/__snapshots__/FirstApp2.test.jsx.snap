// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <FirstApp /> this test must match the snapshot" 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hi MF
     
  </h1>
  <h2>
    No Subtitle
  </h2>
  <h2>
    No Subtitle
  </h2>
  <h2>
    Pabelon
  </h2>
</div>
`;
